src/libraries/Microsoft.PowerFx.Core/Public/Engine.cs

```diff
@@ -124,6 +127,11 @@ private protected INameResolver CreateResolverInternal(out ReadOnlySymbolTable s
             return symbols;
         }
 
+        private protected virtual IExternalRuleScopeResolver CreateExternalRuleScopeResolver()
```

Comment:
This IExternalRuleScopeResolver is used in Canvas App backend when calling TexlBinding.Run(). There was no option to supply a custom implementation of this in Engine, so following the existing pattern in Engine, added this virtual function for the derived classes of Engine to override and supply custom implementation.

Reply:
Add this as a comment in the code! Very useful context

Reply:
Added